### PR TITLE
Lint checks phase 2

### PR DESCRIPTION
**Pull request by SmallCoccinelle**
Next round of lint checks:

* This handles all unchecked errors up to the current HEAD as of this patch. Generally follows the same structure as in errcheck-phase-1: one-liner initializers, unless the line would be excessively long.
* Removes unused functions
* Runs a few simplification fixes

With these patches, the project now passes `golangci-lint run` on the command line, and we can look to enable it parallel to the normal build.

### CODE REVIEW

File: pkg/manager/config/config.go [hunk 1]
```diff
@@ -976,11 +975,23 @@ func (i *Instance) setDefaultValues() error {
...
-// SetInitialConfig fills in missing required config fields
 func (i *Instance) SetInitialConfig() error {
+	return i.setInitialConfig(true)
+}
+
+func (i *Instance) SetInitialMemoryConfig() error {
+	return i.setInitialConfig(false)
+}
+
+// SetInitialConfig fills in missing required config fields
```
Comment (WithoutPants) [inline pkg/manager/config/config.go:993]: This comment should be moved back to its original position above the public method.

**Review by WithoutPants — CHANGES_REQUESTED** [review]
Just a minor thing. Not yet tested.

File: pkg/manager/config/config.go (hunk 1 above)
Comment (SmallCoccinelle) [reply pkg/manager/config/config.go:993]: Oh, darn. I'll push a fix.

File: pkg/database/transaction.go [hunk 2]
```diff
@@ -17,11 +18,15 @@ func WithTxn(fn func(tx *sqlx.Tx) error) error {
 	defer func() {
 		if p := recover(); p != nil {
 			// a panic occurred, rollback and repanic
-			tx.Rollback()
+			if err := tx.Rollback(); err != nil {
+				logger.Warnf("failure when performing tranaction rollback: %v", err)
```
Comment (bnkai) [inline pkg/database/transaction.go:22]: typo `transaction`

File: pkg/database/transaction.go [hunk 3]
```diff
@@ -17,11 +18,15 @@ func WithTxn(fn func(tx *sqlx.Tx) error) error {
...
+			}
 			panic(p)
 		} else if err != nil {
 			// something went wrong, rollback
-			tx.Rollback()
+			if err := tx.Rollback(); err != nil {
+				logger.Warnf("failure when performing tranaction rollback: %v", err)
```
Comment (bnkai) [inline pkg/database/transaction.go:28]: same here `transaction`

File: pkg/ffmpeg/stream.go [hunk 4]
```diff
@@ -32,7 +32,9 @@ func (s *Stream) Serve(w http.ResponseWriter, r *http.Request) {
 	notify := r.Context().Done()
 	go func() {
 		<-notify
-		s.Process.Kill()
+		if err := s.Process.Kill(); err != nil {
+			logger.Warnf("enable to kill os process %v: %v", s.Process.Pid, err)
```
Comment (bnkai) [inline pkg/ffmpeg/stream.go:36]: typo `unable`

File: pkg/database/transaction.go (hunk 3 above)
Comment (SmallCoccinelle) [reply pkg/database/transaction.go]: I've fixed this one, and the one above.

File: pkg/manager/task_generate_markers.go [hunk 5]
```diff
@@ -82,7 +82,9 @@ func (t *GenerateMarkersTask) generateSceneMarkers() {
 
 	// Make the folder for the scenes markers
 	markersFolder := filepath.Join(instance.Paths.Generated.Markers, sceneHash)
-	utils.EnsureDir(markersFolder)
+	if err := utils.EnsureDir(markersFolder); err != nil {
+		logger.Warnf("could not create the marker folder (%v): %v", markersFolder, err)
```
[the remark this answers is not in the document]
Comment (SmallCoccinelle) [reply pkg/manager/task_generate_markers.go]: This too

File: pkg/ffmpeg/stream.go (hunk 4 above)
Comment (SmallCoccinelle) [reply pkg/ffmpeg/stream.go]: Fixed

**Comment by SmallCoccinelle** [comment]
@bnkai Thank you for the fixes!

I've pushed a patch fixing the typos.

**Review by WithoutPants — APPROVED** [review]

**Pull request merged**